### PR TITLE
Add DETR model support for object detection

### PR DESCRIPTION
**Pull request by aaron-siegel**
This change is [Reviewable](https://reviewable.io/reviews/pixeltable/pixeltable/106)

### CODE REVIEW

File: pixeltable/functions/huggingface.py [hunk 1]
```diff
@@ -85,6 +79,28 @@ def clip_image(image: Batch[PIL.Image.Image], *, model_id: str) -> Batch[np.ndar
     return [embeddings[i] for i in range(embeddings.shape[0])]
 
 
+@pxt.udf(batch_size=32)
+def detr_detections(image: Batch[PIL.Image.Image], *, model_id: str, threshold: float = 0.5) -> Batch[dict]:
```
Comment (mkornacker) [inline pixeltable/functions/huggingface.py]: I'm wondering if we should adopt a convention for HF models to use the full model name as the function name, ie, detr_for_object_detection.

File: pixeltable/functions/huggingface.py [hunk 2]
```diff
@@ -85,6 +79,28 @@ def clip_image(image: Batch[PIL.Image.Image], *, model_id: str) -> Batch[np.ndar
...
+    processor = _lookup_processor(model_id, lambda x: DetrImageProcessor.from_pretrained(x, revision='no_timm'))
+
+    inputs = processor(images=image, return_tensors='pt')
+    outputs = model(**inputs)
+
+    results = processor.post_process_object_detection(outputs, threshold=threshold)
+    return [
+        {
+            'scores': [score.item() for score in result['scores']],
+            'labels': [model.config.id2label[label.item()] for label in result['labels']],
```
Comment (mkornacker) [inline pixeltable/functions/huggingface.py]: Wondering if we should also include the ids, might be easier to work with in some scenarios.

File: pixeltable/tests/test_functions.py [hunk 3]
```diff
@@ -106,7 +106,7 @@ def test_gpt_4_vision(self, test_client: pt.Client) -> None:
         t.add_column(response_content=t.response.choices[0].message.content)
         t.insert([{
             'prompt': "What's in this image?",
-            'img': 'https://raw.githubusercontent.com/mkornacker/pixeltable/master/docs/source/data/images/000000000009.jpg'
+            'img': 'https://raw.githubusercontent.com/pixeltable/pixeltable/master/docs/source/data/images/000000000009.jpg'
```
Comment (mkornacker) [inline pixeltable/tests/test_functions.py]: Have something like SAMPLE_IMG_URL In test/utils.py?

**Review by aaron-siegel — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/pixeltable/pixeltable/106)* status: 0 of 2 files reviewed, 3 unresolved discussions (waiting on @mkornacker)

___
*[`pixeltable/functions/huggingface.py` line 83 at r1](https://reviewable.io/reviews/pixeltable/pixeltable/106#gh-1544660351:-NuBdN4A8VqAR6ARCee0:b-896fix) ([raw file](https://github.com/pixeltable/pixeltable/blob/f0fbbaaff5da0dc34e492f09e28e03ad4a791fbb/pixeltable/functions/huggingface.py#L83)):*
<details><summary><i>Previously, mkornacker (Marcel Kornacker) wrote…</i></summary><blockquote>

I'm wondering if we should adopt a convention for HF models to use the full model name as the function name, ie, detr_for_object_detection.
</blockquote></details>

Done.
___
*[`pixeltable/functions/huggingface.py` line 97 at r1](https://reviewable.io/reviews/pixeltable/pixeltable/106#gh-1544661057:-NuA4xk48VbYgxfg1yR2:b-896fix) ([raw file](https://github.com/pixeltable/pixeltable/blob/f0fbbaaff5da0dc34e492f09e28e03ad4a791fbb/pixeltable/functions/huggingface.py#L97)):*
<details><summary><i>Previously, mkornacker (Marcel Kornacker) wrote…</i></summary><blockquote>

Wondering if we should also include the ids, might be easier to work with in some scenarios.
</blockquote></details>

Done.
___
*[`pixeltable/tests/test_functions.py` line 109 at r1](https://reviewable.io/reviews/pixeltable/pixeltable/106#gh-1544661996:-NuBdFLXEAyFNarNZn7F:b-ky4zcv) ([raw file](https://github.com/pixeltable/pixeltable/blob/f0fbbaaff5da0dc34e492f09e28e03ad4a791fbb/pixeltable/tests/test_functions.py#L109)):*
<details><summary><i>Previously, mkornacker (Marcel Kornacker) wrote…</i></summary><blockquote>

Have something like SAMPLE_IMG_URL In test/utils.py?
</blockquote></details>

Done. I just put it in test_functions.py for now.

**Review by mkornacker — APPROVED** [review]

**Pull request merged**